src/app/pages/2_Sequence-Variability.py

```diff
@@ -25,9 +31,30 @@
 
     # Display a button to calculate sequence variability
     if st.button("Calculate Sequence Variability"):
-        if not base_sequence:
-            st.warning("Please enter a base sequence before calculating.")
+        st.session_state[SEQVAR_CALC_BTN] = True
+
+# Button to calculate sequence variability is clicked
+if st.session_state[SEQVAR_CALC_BTN]:
+    if not base_sequence:
+        st.warning("Please enter a base sequence before calculating.")
+    else:
+        # Calculate sequence variability
+        if not st.session_state[SEQVAR_CALCULATED]:
+            st.session_state[SEQVAR_DF] = calculate_sequence_variability(
+                sequences, base_sequence
+            )
+            # Set the flag for calculation of sequence variability to True
+            st.session_state[SEQVAR_CALCULATED] = True
+
+        # Display the sequence variability data
+        if not st.session_state[SEQVAR_DF].empty:
+            st.write("Sequence Variability Data:")
+            st.dataframe(st.session_state[SEQVAR_DF])
+
+            # Enter the species name for downloading the data
+            species = st.text_input("Enter the species name:")
+
+            if st.button("Prepare for Download"):
+                download_seq_var_data(st.session_state[SEQVAR_DF], species)
         else:
-            df = calculate_sequence_variability(sequences, base_sequence)
-            st.write("Sequence Variability Calculation Results:")
-            st.dataframe(df)
+            st.write("No Sequence Variability Data to Download")
```

Comment:
```suggestion
            st.write("No sequence variability data to download")
```

Reply:
Wondering if you need to capitalize on `sequence variability` or not? I feel that it's unnecessary to do so.

Reply:
Yes, looks better without capitalize.